### PR TITLE
Feature/pm 107  previous next subgait

### PR DESCRIPTION
**Pull request by Roelemans**
Closes PM-107
@gaiavdh @marnix have a look if you like the inteface as it is now.

## Description

Added two little menus in the gait generator that allows selecting a previous and a next subgait. The lock start/end point checkbox has not yet been implemented.

## Changes

* Added two menus for selecting previous/next subgait
* Removed unused Publish gait functionality

### CODE REVIEW

File: march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator_controller.py [hunk 1]
```diff
@@ -313,3 +337,41 @@ def redo(self):
     def save_changed_joints(self, joints):
         self.joint_changed_history.append(joints)
         self.joint_changed_redo_list = RingBuffer(capacity=100, dtype=list)
+
+    def toggle_lock_startpoint(self, value):
+        if value and self.previous_subgait is not None:
+            rospy.loginfo('Lock that start to ' + self.previous_subgait.subgait_name)
+
+    def toggle_lock_endpoint(self, value):
+        if value and self.previous_subgait is not None:
```
Comment (Olavhaasie) [inline march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator_controller.py:346]: Got a big red error when I tried to press this button without any next gait loaded:wink:
```suggestion
        if value and self.next_subgait is not None:
```

File: march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator_controller.py [hunk 2]
```diff
@@ -35,6 +38,14 @@ def connect_buttons(self):
         self.view.import_gait_button.clicked.connect(self.import_gait)
         self.view.export_gait_button.clicked.connect(self.export_gait)
 
+        self.view.import_previous_subgait_button.clicked.connect(lambda: self.import_side_subgait('previous'))
+        # self.view.previous_is_standing_check_box.stateChanged.connect(self.toggle_previous_is_standing)
+        self.view.lock_startpoint_check_box.stateChanged.connect(self.toggle_lock_startpoint)
+
+        self.view.import_next_subgait_button.clicked.connect(lambda: self.import_side_subgait('next'))
+        # self.view.previous_is_standing_check_box.stateChanged.connect(self.toggle_next_is_standing)
+        self.view.lock_endpoint_check_box.stateChanged.connect(self.toggle_lock_endpoint)
```
Comment (Olavhaasie) [inline march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator_controller.py:47]: You should remove these commented lines

Comment (RutgerVanBeek) [reply march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator_controller.py:47]: Isn't this necessary for the second part of the issue?

File: march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator_controller.py (hunk 1 above)
Comment (Roelemans) [reply march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator_controller.py:346]: Oopsie

**Comment by Roelemans** [comment]
The standing checkbox is a shortcut when you want to say that the subgait comes from/goes to standing. This is probably the most common previous/next state so I thought it would be nice to create a shortcut for that

**Pull request merged**